pkg/render/apiserver.go

```diff
@@ -321,35 +322,22 @@ func (c *apiServerComponent) Objects() ([]client.Object, []client.Object) {
 		objsToDelete = append(objsToDelete, &admregv1.MutatingWebhookConfiguration{ObjectMeta: metav1.ObjectMeta{Name: common.SidecarMutatingWebhookConfigName}})
 	}
 
-	podSecurityNamespaceLabel := PodSecurityStandard(PSSRestricted)
-	if c.hostNetwork() {
-		podSecurityNamespaceLabel = PSSPrivileged
-	}
-	// Global OSS-only objects.
-	globalCalicoObjects := []client.Object{
-		CreateNamespace(rmeta.APIServerNamespace(operatorv1.Calico), c.cfg.Installation.KubernetesProvider, podSecurityNamespaceLabel, c.cfg.Installation.Azure),
-		CreateOperatorSecretsRoleBinding(rmeta.APIServerNamespace(operatorv1.Calico)),
-	}
-
 	// Compile the final arrays based on the variant.
 	if c.cfg.Installation.Variant == operatorv1.TigeraSecureEnterprise {
 		// Create any enterprise specific objects.
 		globalObjects = append(globalObjects, globalEnterpriseObjects...)
 		namespacedObjects = append(namespacedObjects, namespacedEnterpriseObjects...)
 
-		// Explicitly delete any global OSS objects.
-		// Namespaced objects will be handled by namespace deletion.
-		objsToDelete = append(objsToDelete, globalCalicoObjects...)
-	} else {
-		// Create any Calico-only objects
-		globalObjects = append(globalObjects, globalCalicoObjects...)
+		objsToDelete = append(objsToDelete, c.cleanupDeployment())
```

Comment:
cleanupDeployment ensures cleanup of the calico-apiserver deployment during OSS to EE upgrades, where the API server is replaced by the tigera-apiserver deployment.